_posts/2020-12-02-spack-user-survey-2020.md

```diff
@@ -0,0 +1,918 @@
+their macs, and over half the users within ECP are also running it on
+macOS. A small number of users (< 10%) are running Spack within the
+Windows Subsystem for Linux. We don't test there, but we're told that
+Spack works fine in the WSL environment.
+
+If we look at the responses in more detail, we can see the specific Linux
+*distributions* that users are running. The most popular, by far, are
+CentOS and Red Hat. In ECP, Red Hat is especially popular. Ubuntu is the
+next most popular after these, then macOS, then other Linux distributions
+and WSL. SuSE is more popular within ECP than in the broader community,
+likely because it is the Linux distribution that most Cray systems are
+built on.
+
+
+## How many software installations have you done with Spack in the past year?
+
+<figure>
+  <a href="/assets/images/spack-user-survey-2020/two_multi_bars_num_installations.svg">
+    <img src="/assets/images/spack-user-survey-2020/two_multi_bars_num_installations.svg">
+  </a>
+</figure>
+
+Almost 28% of the community has done over 200 software installations, and
+over 12% have done over 1,000 installs. The ECP numbers are similar to
+the general population, but the distribution is slightly shifted towards
+larger numbers of installations.
+
+## What Python version(s) do you use to run Spack?
+
+<figure>
+  <a href="/assets/images/spack-user-survey-2020/two_multi_bars_python_version.svg">
+    <img src="/assets/images/spack-user-survey-2020/two_multi_bars_python_version.svg">
+  </a>
+</figure>
+
+Support for Python 2 ended January 1, 2020, but nearly a year later,
+around 40% of Spack users are still using Python 2.7. 4 users are even
+using Python 2.6. Python 2.7 is still the system Python version on many
+operating systems, including Red Hat 7 and CentOS 7, and on Red Hat and
+CentOS 6 (which are still used at some sites) the default is 2.6.
+
+While many projects can pick their version of Python, Spack is often the
+tool people use to *install* newer versions of Python, and we don't want
+to make users use another installer just to install Spack's dependencies.
+We want it to work out of the box, so we try to make Spack work with the
+system Python everywhere.
+
+## How bad would it be if Spack dropped support for Python 2.6?
+
+<figure>
+  <a href="/assets/images/spack-user-survey-2020/two_pies_how_bad_no_py26.svg">
+    <img src="/assets/images/spack-user-survey-2020/two_pies_how_bad_no_py26.svg">
+  </a>
+</figure>
+
+We asked whether it would be OK for us to drop Python 2.6, and we found
+that there are still around 4 hold-outs who really need Spack to work on
+Python 2.6, and over 20% of people would be at least mildly
+inconvenienced by this change. For the time being, we'll keep supporting
+Python 2.6, but you can probably expect its deprecation to be announced
+sometime within the next year, as the last few Red Hat 6 installations
+dwindle.
+
+## How bad would it be if Spack only worked with Python 3?
```

Comment:
Not necessarily to the analysis, but it might be interesting to know if Spack's continued support for Python 2 draws users vs EasyBuild. Or perhaps Spack & EasyBuild are comparable in this regard?

Reply:
They're both currently supporting 2.6 and up.  EB has deprecated 2.6 but I don't think it matters here.